gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts

```diff
@@ -332,17 +332,15 @@ export class ApiGeneralInfoComponent implements OnInit, OnDestroy {
         switchMap((policies) =>
           this.matDialog
             .open<GioApiImportDialogComponent, GioApiImportDialogData>(GioApiImportDialogComponent, {
-              data: {
-                apiId: this.apiId,
-                policies,
-              },
+              data: { apiId: this.apiId, policies },
               role: 'alertdialog',
               id: 'importApiDialog',
             })
             .afterClosed(),
         ),
-        filter((confirm) => confirm === true),
-        tap(() => this.ngOnInit()),
+        filter((apiId) => !!apiId),
+        switchMap((apiId: string) => this.apiService.get(apiId)),
```

Comment:
Here we are making a GET request whereas the import method is already returning the updated API.
@ThibaudAV could we modify the import dialog to return the whole API?
I'm asking because I think we have a specific behaviour with the `lastApiFetch$` [here](https://github.com/gravitee-io/gravitee-api-management/blob/8c24e6d0a44c0cec5a36b394e7dca8b32b1ba3d1/gravitee-apim-console-webui/src/services-ngx/api-v2.service.ts#L49) and I don't know if we can rely on the import return

Reply:
🤔 I don't see any other solution than this one. We could improve it so that it only happens at the right time, but since it's code for the V2 APIs, I'm not sure it's that interesting.

Reply:
Hit me up on Slack if you need to. I think we can find a better code here to solve the problem ;)